src/ripple/app/tx/impl/CashCheck.cpp

```diff
@@ -0,0 +1,412 @@
+        {
+            JLOG(ctx.j.warn()) << "Malformed transaction: bad amount: "
+                << value.getFullText();
+            return temBAD_AMOUNT;
+        }
+
+        if (badCurrency() == value.getCurrency())
+        {
+            JLOG(ctx.j.warn()) <<"Malformed transaction: Bad currency.";
+            return temBAD_CURRENCY;
+        }
+    }
+    return preflight2 (ctx);
+}
+
+TER
+CashCheck::preclaim (PreclaimContext const& ctx)
+{
+    auto const sleCheck = ctx.view.read (keylet::check (ctx.tx[sfCheckID]));
+    if (! sleCheck)
+    {
+        JLOG(ctx.j.warn()) << "Check does not exist.";
+        return tecNO_ENTRY;
+    }
+
+    // Only cash a check with this account as the destination.
+    AccountID const dstId {(*sleCheck)[sfDestination]};
+    if (ctx.tx[sfAccount] != dstId)
+    {
+        JLOG(ctx.j.warn()) << "Cashing a check with wrong Destination.";
+        return tecNO_PERMISSION;
+    }
+    AccountID const srcId {(*sleCheck)[sfAccount]};
+    if (srcId == dstId)
+    {
+        // They wrote a check to themselves.  This should be caught when
+        // the check is created, but better late than never.
+        JLOG(ctx.j.warn()) << "Malformed transaction: Check to self.";
+        return temREDUNDANT;
+    }
+    {
+        auto const sleSrc = ctx.view.read (keylet::account (srcId));
+        auto const sleDst = ctx.view.read (keylet::account (dstId));
+        if (!sleSrc || !sleDst)
+        {
+            // If the check exists this should never occur.
+            JLOG(ctx.j.warn())
+                << "Malformed transaction: source or destination not in ledger";
+            return tecNO_ENTRY;
+        }
+
+        if ((sleDst->getFlags() & lsfRequireDestTag) &&
+            !sleCheck->isFieldPresent (sfDestinationTag))
+        {
+            // The tag is basically account-specific information we don't
+            // understand, but we can require someone to fill it in.
+            JLOG(ctx.j.warn())
+                << "Malformed transaction: DestinationTag required in check.";
+            return tecDST_TAG_NEEDED;
+        }
+    }
+    {
+        using duration = NetClock::duration;
+        using timepoint = NetClock::time_point;
+        auto const optExpiry = (*sleCheck)[~sfExpiration];
+
+        // Expiration is defined in terms of the close time of the parent
+        // ledger, because we definitively know the time that it closed but
+        // we do not know the closing time of the ledger that is under
+        // construction.
+        if (optExpiry &&
+            (ctx.view.parentCloseTime() >= timepoint {duration {*optExpiry}}))
+        {
+            JLOG(ctx.j.warn()) << "Cashing a check that has already expired.";
+            return tecEXPIRED;
+        }
+    }
+    {
+        // Preflight verified exactly one of Amount or DeliverMin is present.
+        // Make sure the requested amount is reasonable.
+        STAmount const value {[tx = ctx.tx] ()
```

Comment:
Um, I'll guess, since I'm not in that zone at the moment...

On line 71 I previously needed to create local copies of both `optAmount` and `optDeliverMin` so I could easily determine if they were either both present or both absent.  Since I already had them both in hand it was easy to use the ternary operator.

In contrast, in this location I don't need `optAmount` or the moral equivalent of `optDeliverMin` for any reason besides determining `value`.  So by burying them inside this lambda I allow both:

1. `value` to be `const`, and
2. `optAmount` (or `optDeliverMin`, it didn't matter) to have the smallest possible scope.

If I did not use the lambda either `value` would be non-`const` or `optAmount` would have a larger scope.  I think...

Reply:
This capture actually makes a _copy_ of whatever `ctx.tx` references. Is that the intention?